projects/lights_control/src/main.c

```diff
@@ -3,19 +3,101 @@
 #include "can.h"
 #include "gpio.h"
 #include "interrupt.h"
+#include "log.h"
 #include "status.h"
 #include "wait.h"
 
+#include "lights_blinker.h"
+#include "lights_board_type.h"
+#include "lights_can.h"
+#include "lights_can_config.h"
+#include "lights_gpio.h"
+#include "lights_gpio_config.h"
+#include "lights_gpio_config_front.h"
+#include "lights_signal_fsm.h"
+#include "lights_strobe.h"
+
+#define LIGHTS_SIGNAL_BLINKER_DURATION 500
+#define LIGHTS_SIGNAL_SYNC_COUNT 5
+#define LIGHTS_SIGNAL_NO_SYNC 0
+#define LIGHTS_STROBE_BLINKER_DURATION 500
+
+static LightsSignalFsm s_signal_fsm = { 0 };
+
+static LightsCanStorage s_lights_can_storage = { 0 };
+
+static LightsStrobeStorage s_lights_strobe = { 0 };
+
+static const GPIOAddress s_board_type_address = { .port = GPIO_PORT_B, .pin = 13 };
+
 int main(void) {
-  Event e;
+  CANSettings can_settings = {
+    .bitrate = CAN_HW_BITRATE_500KBPS,
+    .tx = { .port = GPIO_PORT_A, .pin = 12 },
+    .rx = { .port = GPIO_PORT_A, .pin = 11 },
+    .rx_event = LIGHTS_EVENT_CAN_RX,
+    .tx_event = LIGHTS_EVENT_CAN_TX,
+    .fault_event = LIGHTS_EVENT_CAN_FAULT,
+    .loopback = false,
+  };
 
-  // Initialize the various driver control devices
+  // Initialize the libraries.
   gpio_init();
   interrupt_init();
   soft_timer_init();
   event_queue_init();
 
+  // Getting board type.
+  GPIOState state = NUM_GPIO_STATES;
+
+  StatusCode status = gpio_get_state(&s_board_type_address, &state);
+
+  if (!status_ok(status)) {
+    LOG_DEBUG("Error getting board type.\n");
+  }
+
+  LightsBoardType board_type =
+      (state == GPIO_STATE_HIGH) ? LIGHTS_BOARD_TYPE_FRONT : LIGHTS_BOARD_TYPE_REAR;
+  LOG_DEBUG("Board Type: %s\n", (board_type == LIGHTS_BOARD_TYPE_FRONT) ? "Front" : "Back");
+  // Initialize lights_gpio.
+
+  lights_gpio_config_init(board_type);
+
+  LightsGpio *lights_gpio = lights_gpio_config_load();
+  status = lights_gpio_init(lights_gpio);
+
+  if (!status_ok(status)) {
+    LOG_DEBUG("Error initializing lights GPIO.\n");
+  }
+
+  // Initialize lights_can.
+  can_settings.device_id = (board_type == LIGHTS_BOARD_TYPE_FRONT) ? SYSTEM_CAN_DEVICE_LIGHTS_FRONT
+                                                                   : SYSTEM_CAN_DEVICE_LIGHTS_REAR;
+
+  status = lights_can_init(&s_lights_can_storage, lights_can_config_load(), &can_settings);
```

Comment:
You should probably add CAN filters so you don't need to process every single message that's broadcasted on the CAN bus.